lib/connectors/gitlab/connector.rb

```diff
@@ -70,13 +73,13 @@ def custom_client_error
         Connectors::GitLab::CustomClient::ClientError
       end
 
-      def extract_projects
+      def extract_projects(sink)
```

Comment:
Really not a fan of passing `sink` in each method. Can we make it an instance variable?

Reply:
Reverted in https://github.com/elastic/connectors/pull/146/commits/7140895376108bac2542cf07097b3c7607683845

Reply:
Ha, actually it's not possible any more now - because ElasticSink requires knowing where to write data, and for now the way connectors are instantiated they don't have access to connector settings - and it makes sense, because we instatiate a connector also for initial setup, where there are no settings or records in DB, thus it's not possible to actually create a real sink in the initializer, so I decided to move the sink creation to the extraction method, and thus it's not an instance variable any more.

Any thoughts on how to do it better?

Reply:
Agree, but it's not necessary that all instance variables are assigned during instantiation, we can do so when calling `sync_content`